### PR TITLE
Convert t/mojo/daemon.t to use subtests

### PR DESCRIPTION
**Pull request by toshisugimo**
### Summary
- Convert `t/mojo/daemon.t` to use subtests

### Motivation
- Another file converted.

### References
#1520

### CODE REVIEW

File: t/mojo/daemon.t [hunk 1]
```diff
@@ -26,87 +26,100 @@ sub handler {
...
-  ok !!Mojo::Server::Daemon->new->reverse_proxy, 'reverse proxy';
-}
+subtest 'Minimal application' => sub {
+  $ua->server->app(TestApp->new);
+  is $tx->res->code, 200,              'right status';
+  is $tx->res->body, 'Hello TestApp!', 'right content';
+};
+
+subtest 'Timeouts' => sub {
+  {
```
Comment (kraih) [inline t/mojo/daemon.t:41]: Why the block?

File: t/mojo/daemon.t [hunk 2]
```diff
@@ -26,87 +26,100 @@ sub handler {
...
-# Listen
-{
-  is_deeply(Mojo::Server::Daemon->new->listen, ['http://*:3000'], 'right value');
-  local $ENV{MOJO_LISTEN} = 'http://127.0.0.1:8080';
-  is_deeply(Mojo::Server::Daemon->new->listen, ['http://127.0.0.1:8080'], 'right value');
-  $ENV{MOJO_LISTEN} = 'http://*:80,https://*:443';
-  is_deeply(Mojo::Server::Daemon->new->listen, ['http://*:80', 'https://*:443'], 'right value');
-}
+my $ua     = Mojo::UserAgent->new;
+my $tx     = $ua->get('/');
```
Comment (kraih) [inline t/mojo/daemon.t:30]: I don't think this line belongs here.

File: t/mojo/daemon.t [hunk 3]
```diff
@@ -155,191 +168,210 @@ $app->routes->any(
...
+    }
+  );
+  $loop->client({port => $acceptor->port} => sub { }) for 1 .. 2;
+  $loop->start;
+  ok $accepting[0], 'accepting connections';
+  ok !$accepting[1], 'connection limit reached';
+};
+
+subtest 'Request limit' => sub {
+  $daemon = Mojo::Server::Daemon->new(app => $app, listen => ['http://127.0.0.1'], silent => 1)->start;
```
Comment (kraih) [inline t/mojo/daemon.t:342]: `$daemon` is not in the right scope.

File: t/mojo/daemon.t [hunk 4]
```diff
@@ -155,191 +168,210 @@ $app->routes->any(
...
+  is $tx->res->code, 200, 'right status';
+  is $tx->res->body, $port, 'right content';
+};
+
+subtest 'No TLS support' => sub {
+  eval { Mojo::Server::Daemon->new(listen => ['https://127.0.0.1'], silent => 1)->start };
+  like $@, qr/IO::Socket::SSL/, 'right error';
+};
+
+subtest 'bstract methods' => sub {
```
Comment (kraih) [inline t/mojo/daemon.t:372]: Typo.

**Comment by toshisugimo** [comment]
56a93d9...0c444f9
I fixed the places that were pointed out.

File: t/mojo/daemon.t (hunk 1 above)
Comment (toshisugimo) [reply t/mojo/daemon.t]: It was unnecessary .

**Comment by kraih** [comment]
Squash your commits.

**Comment by toshisugimo** [comment]
Squashed

**Review by christopherraa — APPROVED** [review]
This looks good to me. :+1:

**Review by marcusramberg — APPROVED** [review]
LGTM 👍

**Comment by kraih** [comment]
You missed these issues. https://github.com/mojolicious/mojo/compare/34d86b6fadcc...46b8784d7c4f

**Comment by toshisugimo** [comment]
Oops. Thank you for fixing it.

**Pull request merged**